source/MaterialXGenGlsl/Nodes/SurfaceNodeGlsl.cpp

```diff
@@ -233,4 +207,39 @@ void SurfaceNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& conte
     END_SHADER_STAGE(stage, Stage::PIXEL)
 }
 
+void SurfaceNodeGlsl::emitLightLoop(const ShaderNode& node, GenContext& context, ShaderStage& stage, const string& outColor) const
+{
+    const GlslShaderGenerator& shadergen = static_cast<const GlslShaderGenerator&>(context.getShaderGenerator());
+
+    const ShaderGraph& graph = *node.getParent();
```

Comment:
Nitpick: `shadergen` and `graph` is not used outside the if, so they could be moved inside the if-statement.